File: Source/cardgame-client/src/app/domain/game/card/Countess.ts
import { Card } from './card';
import { CardStrength } from './CardStrength';

export class Countess extends Card {
    public GetMustPlay(other?: CardStrength): boolean {
        if (!other) { return false; }
        return other === CardStrength.King || other === CardStrength.Prince;
    }
}

File: Source/CardGame.Domain/Card/Countess.cs
using CardGame.Domain.Abstractions.Card;
using CardGame.Utils.Factory;

namespace CardGame.Domain.Card
{
    public class Countess : Card
    {
        protected Countess(CardId cardId) : base(cardId)
        {
        }
        public static FactoryResult<Card> Factory(int variant)
        {
            var result = CardId.Factory(CardStrength.Countess, variant);
            if (result.IsError)
            {
                return FactoryResult<Card>.Error(result.ErrorMessage);
            }
            return FactoryResult<Card>.Success(new Countess(result.Value));
        }
    }
}